src/Ens.ts

```diff
@@ -0,0 +1,638 @@
+    const nodeHash = this.namehash(domain);
+    const resolverAddr = await this.callMethod(
+      this.registryContract,
+      'resolver',
+      [nodeHash],
+    );
+    if (isNullAddress(resolverAddr)) {
+      throw new ResolutionError(ResolutionErrorCode.UnspecifiedResolver);
+    }
+    return resolverAddr;
+  }
+
+  async record(domain: string, key: string): Promise<string> {
+    const returnee = await this.getTextRecord(domain, key);
+    if (!returnee) {
+      throw new ResolutionError(ResolutionErrorCode.RecordNotFound, {
+        domain,
+        recordName: key,
+        method: this.name,
+        methodName: 'record',
+      });
+    }
+
+    return returnee;
+  }
+
+  async records(
+    domain: string,
+    keys: string[],
+  ): Promise<Record<string, string>> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      methodName: 'records',
+      domain,
+      method: this.serviceName(),
+    });
+  }
+
+  // TODO: Need to understand how ENS supports DNS @see https://docs.ens.domains/contract-api-reference/dns-registrar#retrieving-the-dns-text-record
+  async dns(domain: string, types: DnsRecordType[]): Promise<DnsRecord[]> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      methodName: 'dns',
+      method: this.serviceName(),
+      domain,
+    });
+  }
+
+  async reverse(
+    address: string,
+    currencyTicker: string,
+  ): Promise<string | null> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      methodName: 'reverse',
+    });
+  }
+
+  // TODO: Figure out why nameHash() does not work for reverse address.
+  // Current implementation uses reverseRegistrarContract to fetch the correct node hash.
+  // @see: https://eips.ethereum.org/EIPS/eip-181
+  async reverseOf(address: string): Promise<string | null> {
+    const originalAddress = address;
+    if (address.startsWith('0x')) {
+      address = address.substr(2);
+    }
+
+    const reverseRegistrarAddress = this.determineReverseRegistrarAddress(
+      this.network,
+    );
+    const reverseRegistrarContract = new EthereumContract(
+      reverseRegistrarInterface,
+      reverseRegistrarAddress,
+      this.provider,
+      this.proxyServiceApiKey,
+    );
+
+    const nodeHash = await this.reverseRegistrarCallToNode(
+      reverseRegistrarContract,
+      address,
+    );
+    const resolverAddress = await this.callMethod(
+      this.registryContract,
+      'resolver',
+      [nodeHash],
+    );
+
+    if (isNullAddress(resolverAddress)) {
+      return null;
+    }
+
+    const resolverContract = new EthereumContract(
+      resolverInterface(resolverAddress, EthCoinIndex),
+      resolverAddress,
+      this.provider,
+      this.proxyServiceApiKey,
+    );
+
+    const domainName = await this.resolverCallToName(
+      resolverContract,
+      nodeHash,
+    );
+    const fetchedAddress = await this.addr(domainName, BlockchainType.ETH);
```

Comment:
this should use the addr(namehash) version, not the addr(namehash, coinType). you're using the addr.60 to check which may or may not be available